Rakefile

```diff
@@ -36,3 +36,64 @@ task :gen_nodeset do
   end
   puts nodeset
 end
+
+require 'rspec/core/rake_task'
+desc 'test tiering'
+RSpec::Core::RakeTask.new(:test_tier) do |t|
+
+  # Test mode must always be set
+  test_mode = ENV['BEAKER_TESTMODE'].to_sym
+  raise 'BEAKER_TESTMODE env variable must be either agent or apply.' unless %w(agent apply).include?(test_mode.to_s)
+
+  # If no hosts file is set, use beaker-hostgenerator to build one based on env variables
+  # BEAKER_MASTER & BEAKER_AGENT env variables accept valid beaker-hostgenerator platforms
+  # test_mode = agent
+  #   PE_DIR        - required.
+  #   BEAKER_MASTER - required.
+  #   BEAKER_AGENT  - optional - if not set this will run master and agent on one VM.
+  # test_mode = apply
+  #   BEAKER_AGENT  - required.
+  if(!ENV['BEAKER_setfile'] and !ENV['BEAKER_set'])
```

Comment:
Should this be that instead of raising errors when incorrect environment variables are set to use beaker host generator, it falls through to default, or would you remove beaker host generator all together?